src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DisplayMetadata.cs

```diff
@@ -40,10 +41,10 @@ public class DisplayMetadata
         public string DisplayFormatString { get; set; }
 
         /// <summary>
-        /// Gets or sets a display name for the model.
+        /// Gets or sets a delegate used for getting the display name for the model.
         /// See <see cref="ModelMetadata.DisplayName"/>
         /// </summary>
-        public string DisplayName { get; set; }
+        public Func<string, string> DisplayName { get; set; }
```

Comment:
BTW this isn't about performance. It's about DRY, providing one way to do something.